<a href="https://colab.research.google.com/github/AlmasMalik66/DataScience-AI-Assignments/blob/main/Week2/Datacollection_%26_Cleaning.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Step 1: Load Dataset**

In [1]:
import pandas as pd

# Load your Fake News dataset
df = pd.read_csv('/Fake.csv')

# Show first 5 rows
df.head()


title  \
0   Donald Trump Sends Out Embarrassing New Year’...   
1   Drunk Bragging Trump Staffer Started Russian ...   
2   Sheriff David Clarke Becomes An Internet Joke...   
3   Trump Is So Obsessed He Even Has Obama’s Name...   
4   Pope Francis Just Called Out Donald Trump Dur...   

                                                text subject  \
0  Donald Trump just couldn t wish all Americans ...    News   
1  House Intelligence Committee Chairman Devin Nu...    News   
2  On Friday, it was revealed that former Milwauk...    News   
3  On Christmas day, Donald Trump announced that ...    News   
4  Pope Francis used his annual Christmas Day mes...    News   

                date  
0  December 31, 2017  
1  December 31, 2017  
2  December 30, 2017  
3  December 29, 2017  
4  December 25, 2017

# **Step 2: Check Dataset Information**

In [3]:
# Shape (rows & columns)
print("Shape:", df.shape)

Shape: (23481, 4)


In [4]:
# Data types & missing values
print("\nInfo:")
df.info()


Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23481 entries, 0 to 23480
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   title    23481 non-null  object
 1   text     23481 non-null  object
 2   subject  23481 non-null  object
 3   date     23481 non-null  object
dtypes: object(4)
memory usage: 733.9+ KB


In [6]:
# Summary stats
print("\nDescribe:")
df.describe(include='all')


Describe:


title   text subject  \
count                                               23481  23481   23481   
unique                                              17903  17455       6   
top     MEDIA IGNORES Time That Bill Clinton FIRED His...           News   
freq                                                    6    626    9050   

                date  
count          23481  
unique          1681  
top     May 10, 2017  
freq              46

# **Step 3: Check for Missing Values**

In [8]:
# Missing values count
print("\nMissing values per column:")
df.isnull().sum().sort_values(ascending=False)


Missing values per column:


title      0
text       0
subject    0
date       0
dtype: int64

# **Step 4: Remove duplicates**

In [21]:
Rduplicate = df.duplicated().sum()
print(f"Number of duplicate rows: {Rduplicate}")

Number of duplicate rows: 0


# **Check Missing Values**

In [24]:
missingVal = df.isnull().sum().sum()
print(f"Missing value in DATASET:{missingVal}")

Missing value in DATASET:0


In [25]:
print(df.isnull().sum())

title      0
text       0
subject    0
date       0
dtype: int64


# **Step 5: Detect “Outlier-like” Text Lengths**

In [26]:
# Check the length of text column
df['text_length'] = df['text'].apply(len)

print("Average text length:", df['text_length'].mean())
print("Maximum text length:", df['text_length'].max())
print("Minimum text length:", df['text_length'].min())

# Optional: identify very short texts (potential outliers)
short_texts = df[df['text_length'] < 50]
print("Short text articles:", len(short_texts))


Average text length: 2547.5558821023938
Maximum text length: 51794
Minimum text length: 1
Short text articles: 835


# **Step 6: Remove Outlier-like Short Articles**

In [27]:
# Remove short/incomplete text articles
df = df[df['text_length'] > 50]

# Check new shape
print("Shape after cleaning:", df.shape)


Shape after cleaning: (22642, 5)


In [28]:
df

title  \
0       Donald Trump Sends Out Embarrassing New Year’...   
1       Drunk Bragging Trump Staffer Started Russian ...   
2       Sheriff David Clarke Becomes An Internet Joke...   
3       Trump Is So Obsessed He Even Has Obama’s Name...   
4       Pope Francis Just Called Out Donald Trump Dur...   
...                                                  ...   
23476  McPain: John McCain Furious That Iran Treated ...   
23477  JUSTICE? Yahoo Settles E-mail Privacy Class-ac...   
23478  Sunnistan: US and Allied ‘Safe Zone’ Plan to T...   
23479  How to Blow $700 Million: Al Jazeera America F...   
23480  10 U.S. Navy Sailors Held by Iranian Military ...   

                                                    text      subject  \
0      Donald Trump just couldn t wish all Americans ...         News   
1      House Intelligence Committee Chairman Devin Nu...         News   
2      On Friday, it was revealed that former Milwauk...         News   
3      On Christmas day, Donald Trump announced that ...         News   
4      Pope Francis used his annual Christmas Day mes...         News   
...                                                  ...          ...   
23476  21st Century Wire says As 21WIRE reported earl...  Middle-east   
23477  21st Century Wire says It s a familiar theme. ...  Middle-east   
23478  Patrick Henningsen  21st Century WireRemember ...  Middle-east   
23479  21st Century Wire says Al Jazeera America will...  Middle-east   
23480  21st Century Wire says As 21WIRE predicted in ...  Middle-east   

                    date  text_length  
0      December 31, 2017         2893  
1      December 31, 2017         1898  
2      December 30, 2017         3597  
3      December 29, 2017         2774  
4      December 25, 2017         2346  
...                  ...          ...  
23476   January 16, 2016         3237  
23477   January 16, 2016         1684  
23478   January 15, 2016        25065  
23479   January 14, 2016         2685  
23480   January 12, 2016         5251  

[22642 rows x 5 columns]

# **Before vs After Cleaning**

| Feature | Before Cleaning | After Cleaning |
|----------|----------------|----------------|
| Shape | (23481, 4) | (22646, 5) |
| Missing Values | 0 | 0 |
| Duplicates | 0 | 0 |
| Short / Invalid Texts | 835 short rows | Removed |
| Columns | title, text, subject, date | + text_length added |
